src/SurveyService.Validation/Question/CreateQuestionRequestValidator.cs

```diff
@@ -0,0 +1,52 @@
+﻿using FluentValidation;
+using LT.DigitalOffice.SurveyService.Data.Interfaces;
+using LT.DigitalOffice.SurveyService.Models.Dto.Requests.Question;
+using LT.DigitalOffice.SurveyService.Models.Dto.Requests.Question.Filters;
+using LT.DigitalOffice.SurveyService.Validation.Option.Interfaces;
+using LT.DigitalOffice.SurveyService.Validation.Question.Interfaces;
+using System;
+
+namespace LT.DigitalOffice.SurveyService.Validation.Question;
+
+public class CreateQuestionRequestValidator : AbstractValidator<CreateSingleQuestionRequest>, ICreateQuestionRequestValidator
+{
+  public CreateQuestionRequestValidator(
+    IQuestionRepository _questionRepository
+    )
+  {
+    RuleFor(q => q.Content)
+      .MaximumLength(300)
+      .WithMessage("Content is too long.");
+
+    When(q => q.Deadline.HasValue, () =>
+    {
+      RuleFor(q => q.Deadline)
+        .Must(d => d > DateTime.UtcNow.AddDays(1).AddSeconds(2))
+        .WithMessage("The deadline must be at least 24 hours from now.");
+    });
+
+    When(q => q.GroupId.HasValue, () =>
+    {
+      RuleFor(q => q)
+        .MustAsync(async (q, _) => (await _questionRepository.GetPropertiesAsync(new GetQuestionPropertiesFilter() { GroupId = q.GroupId })).Deadline == q.Deadline)
+        .WithMessage("The deadlines of questions in the group are not equal.");
+
+      RuleFor(q => q)
+        .MustAsync(async (q, _) => (await _questionRepository.GetPropertiesAsync(new GetQuestionPropertiesFilter() { GroupId = q.GroupId })).HasRealTimeResult == q.HasRealTimeResult)
+        .WithMessage("The conditions for displaying the results of the questions are different.");
+    });
+
+    When(q =>  !q.HasCustomOptions, () =>
+    {
+      RuleFor(q => q.Options)
+        .NotEmpty().WithMessage("No options in the question.");
```

Comment:
please, move '.WithMessage..' to next line
and why not "This question should have one option at least." in message here?

Reply:
Done